src/bluesearch/entrypoint/database/topic_extract.py

```diff
@@ -0,0 +1,180 @@
+        """,
+    )
+    parser.add_argument(
+        "-m",
+        "--match-filename",
+        type=str,
+        help="""
+        Extract topic only of articles with a name matching the given regular
+        expression. Ignored when 'input_path' is a path to a file.
+        """,
+    )
+    parser.add_argument(
+        "-R",
+        "--recursive",
+        action="store_true",
+        help="""
+        Find articles recursively.
+        """,
+    )
+    parser.add_argument(
+        "-o",
+        "--overwrite",
+        action="store_true",
+        help="""
+        If output_file exists and overwrite is true, the output file is overwrote.
+        """,
+    )
+    return parser
+
+
+def run(
+    *,
+    input_source: str,
+    input_path: Path,
+    output_file: Path,
+    match_filename: str,
+    recursive: bool,
+    overwrite: bool,
+) -> int:
+    """Extract topic of articles.
+
+    Parameter description and potential defaults are documented inside of the
+    `get_parser` function.
+    """
+    import json
+
+    import bluesearch
+    from bluesearch.database.topic import get_topics_for_pmc_article
+
+    inputs: Iterable[Path]
+
+    if input_path.is_file():
+        inputs = [input_path]
+
+    elif input_path.is_dir():
+        if recursive:
+            pattern = "**/*"
+        else:
+            pattern = "*"
+        files = (x for x in input_path.glob(pattern) if x.is_file())
+
+        if match_filename is None:
+            selected = files
+        elif match_filename == "":
+            raise ValueError("Value for argument 'match-filename' should not be empty!")
+        else:
+            regex = re.compile(match_filename)
+            selected = (x for x in files if regex.fullmatch(x.name))
+
+        inputs = sorted(selected)
+
+    else:
+        raise ValueError(
+            "Argument 'input_path' should be a path to an existing file or directory!"
+        )
+
+    if output_file.exists() and not overwrite:
+        with open(output_file) as f:
+            all_results = json.load(f)
+    else:
+        all_results = []
+
+    if input_source == "pmc":
+        for path in inputs:
+            journal_topics, article_topics = get_topics_for_pmc_article(path)
+            all_results.append(
+                {
+                    "source": "pmc",
+                    "path": path,
+                    "topics": {
+                        "article": {
+                            "MeSH": article_topics,
+                        },
+                        "journal": {
+                            "MeSH": journal_topics,
+                        },
+                    },
+                    "metadata": {
+                        "created-date": datetime.datetime.now(),
+                        "bbs-version": bluesearch.version,
+                    },
+                }
+            )
+
+    with open(output_file, "wb") as f:
+        json.dump(all_results, f)
```

Comment:
Why do we want to write it as binary? We want the `.jsonl` to be a readable text file, right?

Reply:
Also, it would be nice to do `sort_keys=True, indent=4` or something similar.

Reply:
Actually, we can use `bluesearch.utils.JSONL` for both loading and dumping